File: order-service/src/main/java/com/example/orderservice/domain/OrderItem.java
package com.example.orderservice.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import static jakarta.persistence.FetchType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
public class OrderItem {

    @Id
    @GeneratedValue
    @Column(name = "order_item_id")
    private Long id;

    @ManyToOne(fetch = LAZY)
    private Order order;

    private String name;

    private String itemUUID;

    private int quantity;

    private int price;

    @Builder
    public OrderItem(String name, String itemUUID, int quantity, int price) {
        this.name = name;
        this.itemUUID = itemUUID;
        this.quantity = quantity;
        this.price = price;
    }

    public void linkOrder(Order order) {
        this.order = order;
    }


}

File: front/src/entity/order/OrderItem.ts
import type ShoppingCartItem from "@/entity/item/ShoppingCartItem";

export default class OrderItem {
    public name = ''
    public itemUUID = ''
    public quantity = 0
    public price = 0


    constructor(cartItem: ShoppingCartItem) {
        if (cartItem) {
            this.name = cartItem.name;
            this.itemUUID = cartItem.itemUUID;
            this.quantity = cartItem.quantity;
            this.price = cartItem.itemPrice;
        }
    }


}